include/net/ethernet.h

```diff
@@ -135,7 +135,7 @@ struct ethernet_api {
 
 #if defined(CONFIG_PTP_CLOCK)
 	/** Return ptp_clock device that is tied to this ethernet device */
-	struct device *(*get_ptp_clock)(struct device *eth_dev);
+	struct device *(*get_ptp_clock)(struct device *dev);
```

Comment:
I disagree here. The purpose of the eth_dev name was to emphasize that the dev parameter is the actual ethernet device, and not the ptp clock device. Otherwise this can be very confusing to the reader.

Reply:
All functions of a device driver API should take the actual device you are talking to. Obviously you are not going to request get_ptp_clock from a ptp clock device, that's makes no sense. Plus the comment you put right before explains that already. Let's juste use dev everywhere for the first param, as any other device driver api do (minor the ones who have a name as "port" like gpio)

Reply:
I guess it's minor point after all. If @jukkar wanted the param to be self-documenting by being eth_dev, maybe it makes sense to leave it such.

Reply:
no it does not. Either is foo bar for all the device driver api, but not both or else in the middle.

It's not because ethernet device driver api is not in include but include/net/ that it should not follow the same rules as all the device driver api.

Reply:
Hmm, I am trying to make these things self documenting but it is like hitting a head to brick wall. These various rules are becoming really pedantic imho. Anyway, lets merge this then, no need to spend so much time arguing.